indexing-service/src/main/java/io/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -299,10 +358,25 @@ public TaskStatus call()
                               if (taskWorkItem != null && taskWorkItem.processHolder != null) {
                                 taskWorkItem.processHolder.process.destroy();
                               }
+                              if (!stopping) {
+                                saveRestorableTasks();
+                              }
                             }
+
                             portFinder.markPortUnused(childPort);
-                            log.info("Removing temporary directory: %s", attemptDir);
-                            FileUtils.deleteDirectory(attemptDir);
+
+                            try {
+                              if (!stopping && taskDir.exists()) {
+                                log.info("Removing task directory: %s", taskDir);
+                                FileUtils.deleteDirectory(taskDir);
+                              }
+                            }
+                            catch (Exception e) {
```

Comment:
Blanket exception catches to not handle InterruptedException quite right if you're wanting to retain the thread's Interrupted status.